src/transformers/models/instructblip/processing_instructblip.py

```diff
@@ -205,7 +204,17 @@ def save_pretrained(self, save_directory, **kwargs):
         os.makedirs(save_directory, exist_ok=True)
         qformer_tokenizer_path = os.path.join(save_directory, "qformer_tokenizer")
         self.qformer_tokenizer.save_pretrained(qformer_tokenizer_path)
-        return super().save_pretrained(save_directory, **kwargs)
+
+        # We modify the attributes so that only the tokenizer and image processor are saved in the main folder
+        qformer_present = "qformer_tokenizer" in self.attributes
+        if qformer_present:
+            self.attributes.remove("qformer_tokenizer")
+
+        outputs = super().save_pretrained(save_directory, **kwargs)
+
+        if qformer_present:
+            self.attributes += ["qformer_tokenizer"]
+        return outputs
```

Comment:
Out of curiosity, why don't we want the qformer_tokenizer to be saved here?

Reply:
What happens in `processor.save_pretrained` is that it iterates over the classes listed in `processor.attributes`. In the case of `qformer_tokenizer`, if it was saved out, its tokenizer files e.g. `tokenizer.json` would just overwrite the files of the `tokenizer` class earlier in the list. To avoid this, a workaround was done, such that the `qformer_tokenizer` was saved to a subfolder e.g. [like here](https://huggingface.co/Salesforce/instructblip-vicuna-7b/tree/main/qformer_tokenizer), and just the tokenizer was saved to the top-level of the checkpoint.